src/js/utils/strings.ts

```diff
@@ -40,23 +40,24 @@ export function extension(path) {
     if (path.lastIndexOf('.') > -1) {
         return path.substr(path.lastIndexOf('.') + 1, path.length).toLowerCase();
     }
+    return '';
 }
 
 // Convert seconds to HH:MN:SS.sss
-export function hms(secondsNumber) {
+export function hms(secondsNumber: number): string {
     const h = (secondsNumber / 3600) | 0;
     const m = ((secondsNumber / 60) | 0) % 60;
     const s = secondsNumber % 60;
-    return pad(h, 2) + ':' + pad(m, 2) + ':' + pad(s.toFixed(3), 6);
+    return pad(h.toString(), 2) + ':' + pad(m.toString(), 2) + ':' + pad(s.toFixed(3), 6);
 }
 
 // Convert a time-representing string to a number
-export function seconds(str, frameRate) {
+export function seconds(str: string, frameRate?: number): number {
     if (!str) {
```

Comment:
We should either remove the if(!str) check, or make str nullable

Reply:
Made it an optional param. Just to note, I do agree with what you said where we may need to revisit our optional params in the future.

Reply:
Made it an optional param. Just to note, I do agree with what you said where we may need to revisit our optional params in the future.